File: mcspring-api/mcspring-base/src/main/java/in/kyle/mcspring/command/registration/CommandController.java
package in.kyle.mcspring.command.registration;

import org.bukkit.command.Command;
import org.bukkit.command.CommandMap;
import org.bukkit.plugin.Plugin;
import org.springframework.stereotype.Controller;

import in.kyle.mcspring.RequiresSpigot;
import lombok.RequiredArgsConstructor;

@Controller
@RequiredArgsConstructor
@RequiresSpigot
class CommandController {
    
    private final CommandMap commandMap;
    private final Plugin plugin;
    
    public void registerCommand(Command command) {
        commandMap.register(command.getLabel(), plugin.getName(), command);
    }
}

File: mcspring-api/mcspring-base/src/main/kotlin/in/kyle/mcspring/command/registration/CommandController.kt
package `in`.kyle.mcspring.command.registration

import `in`.kyle.mcspring.common.commands.CommandMapWrapper
import org.bukkit.command.Command
import org.bukkit.plugin.Plugin
import org.springframework.boot.autoconfigure.condition.ConditionalOnBean
import org.springframework.context.annotation.Bean
import org.springframework.stereotype.Controller

@Controller
@ConditionalOnBean(Plugin::class)
internal class CommandController(
        private val plugin: Plugin
) {
    fun registerCommand(command: Command) = CommandMapWrapper.registerCommand(plugin, command)

    @Bean
    fun commandMap() = CommandMapWrapper.commandMap
}
